disttask/framework/dispatcher/dispatcher.go

```diff
@@ -111,35 +115,43 @@ type dispatcher struct {
 }
 
 // NewDispatcher creates a dispatcher struct.
-func NewDispatcher(ctx context.Context, taskTable *storage.TaskManager) (Dispatch, error) {
+func NewDispatcher(ctx context.Context, taskTable *storage.TaskManager) (Dispatcher, error) {
 	dispatcher := &dispatcher{
 		taskMgr:              taskTable,
 		detectPendingGTaskCh: make(chan *proto.Task, DefaultDispatchConcurrency),
 	}
-	pool, err := spool.NewPool("dispatch_pool", int32(DefaultDispatchConcurrency), util.DistTask, spool.WithBlocking(true))
+	gPool, err := spool.NewPool("dispatch_pool", int32(DefaultDispatchConcurrency), util.DistTask, spool.WithBlocking(true))
 	if err != nil {
 		return nil, err
 	}
-	dispatcher.gPool = pool
+	dispatcher.gPool = gPool
 	dispatcher.ctx, dispatcher.cancel = context.WithCancel(ctx)
 	dispatcher.runningGTasks.taskIDs = make(map[int64]struct{})
 
 	return dispatcher, nil
 }
 
-// Start implements Dispatch.Start interface.
+// Start implements dispatcher.Start interface.
 func (d *dispatcher) Start() {
 	d.wg.Run(d.DispatchTaskLoop)
-	d.wg.Run(d.DetectTaskLoop)
+	d.inited = true
 }
 
-// Stop implements Dispatch.Stop interface.
+// Stop implements dispatcher.Stop interface.
 func (d *dispatcher) Stop() {
 	d.cancel()
 	d.gPool.ReleaseAndWait()
 	d.wg.Wait()
+	d.inited = false
 }
 
+func (d *dispatcher) Inited() bool {
```

Comment:
We don't use this function. Could we remove it?

Reply:
I think `Inited` is better for checking of the dispatcher is started.